component-generator-ui/src/reducers/ComponentGeneratorDataReducer.js

```diff
@@ -0,0 +1,479 @@
+                    propertiesTabUpdate[index].label = tabToUpdate.label;
+                }
+                propertiesTabUpdate[index].fields = tabToUpdate.fields;
+            }
+        });
+        return {
+            ...state.options,
+            propertiesGlobalTabs: propertiesTabUpdate,
+        };
+    }
+    default:
+        return {
+            ...state.options,
+        };
+    }
+}
+
+function propertiesAdder(state, propToAdd) {
+    return {
+        ...state.options,
+        properties: state.options.properties.concat(propToAdd),
+    };
+}
+
+function propertiesSharedAdder(state, propToAdd) {
+    return {
+        ...state.options,
+        propertiesShared: state.options.propertiesShared.concat(propToAdd),
+    };
+}
+
+function propertiesGlobalAdder(state, propToAdd) {
+    return {
+        ...state.options,
+        propertiesGlobal: state.options.propertiesGlobal.concat(propToAdd),
+    };
+}
+
+function tabsAdder(state, tabToAdd) {
+    const { type } = tabToAdd;
+    switch (type) {
+    case MAIN: {
+        return {
+            ...state.options,
+            propertiesTabs: state.options.propertiesTabs.concat(tabToAdd),
+        };
+    }
+    case SHARED: {
+        return {
+            ...state.options,
+            propertiesSharedTabs: state.options.propertiesSharedTabs.concat(tabToAdd),
+        };
+    }
+    case GLOBAL: {
+        return {
+            ...state.options,
+            propertiesGlobalTabs: state.options.propertiesGlobalTabs.concat(tabToAdd),
+        };
+    }
+    default:
+        return {
+            ...state.options,
+        };
+    }
+}
+
+function propertiesMover(state, propToMove) {
+    return {
+        ...state.options,
+        properties: arrayMove(state.options.properties, propToMove.oldIndex, propToMove.newIndex),
+    };
+}
+
+function propertiesMoverShared(state, propToMove) {
+    return {
+        ...state.options,
+        propertiesShared: arrayMove(state.options.propertiesShared, propToMove.oldIndex, propToMove.newIndex),
+    };
+}
+
+function propertiesMoverGlobal(state, propToMove) {
+    return {
+        ...state.options,
+        propertiesGlobal: arrayMove(state.options.propertiesGlobal, propToMove.oldIndex, propToMove.newIndex),
+    };
+}
+
+function tabsMover(state, tabToMove) {
+    return {
+        ...state.options,
+        propertiesTabs: arrayMove(state.options.propertiesTabs, tabToMove.oldIndex, tabToMove.newIndex),
+    };
+}
+
+export default function (state = INITIAL_STATE, action) {
+    switch (action.type) {
+    case FETCH_CONFIGS:
+        return {
+            ...state,
+            codeOwner: action.payload['project-settings']['code-owner'],
+            bundlePath: action.payload['project-settings']['bundle-path'],
+            testPath: action.payload['project-settings']['test-path'],
+            appsPath: action.payload['project-settings']['apps-path'],
+            copyrightYear: action.payload['project-settings'].year,
```

Comment:
Let's please restore this to `copyright-year` (here and everywhere) for backward compatibiilty